stubs/opencv-python/cv2/gapi/__init__.pyi

```diff
@@ -0,0 +1,107 @@
+from _typeshed import Self
+from collections.abc import Callable, Iterable, Sequence
+from typing import Protocol, TypeVar
+
+from cv2.cv2 import GArrayT, GCompileArg, GOpaqueT, gapi_GKernelPackage, gapi_GNetPackage, gapi_ie_PyParams
+from cv2.gapi.streaming import queue_capacity
+
+class _KernelCls(Protocol):
+    id: str
+    outMeta: object
+    on: staticmethod[tuple[type, ...] | type]
+
+_K = TypeVar("_K", bound=_KernelCls)
+_F = TypeVar("_F", bound=Callable[..., object])
+_A = TypeVar("_A")
+
+def register(mname: str) -> Callable[[_F], _F]: ...
+def networks(*args: gapi_ie_PyParams) -> gapi_GNetPackage: ...
+def compile_args(*args: gapi_GKernelPackage | gapi_GNetPackage | queue_capacity) -> list[GCompileArg]: ...
+def GIn(*args: _A) -> list[_A]: ...
+def GOut(*args: _A) -> list[_A]: ...
+def gin(*args: _A) -> list[_A]: ...
+def descr_of(*args: _A) -> list[_A]: ...
+
+class GOpaque:
+    def __new__(cls, argtype: int) -> GOpaqueT: ...  # type: ignore[misc]
```

Comment:
What are all the `type: ignore`s here for?

Reply:
mypy is not happy that `__new__` doesn't return a subtype of `GOpaque`.
I'm adding the following comment from source to make it clearer why it's like this:
```py
# NB: Inheritance from c++ class cause segfault.
# So just aggregate cv.GOpaqueT instead of inheritance
```

Reply:
I think it might be slightly clearer if you added a comment to the effect of "the `type: ignore`s are because mypy doesn't like `__new__` returning something that isn't a subtype of the class" as well :)